### PR TITLE
refactor(connector): [Square] change error message from NotSupported to NotImplemented

### LINKED ISSUE #2861: [REFACTOR] : [Square] Error Message For Connector Implementation
### :memo: Feature Description

- In terms of errors, we are currently throwing a 'not supported' message, which shouldn't be the case as we are yet to decide on the implementation.

### :hammer: Possible Implementation

- In order to manage payment methods that are not implemented by Hyperswitch or yet to be implemented, the connector transformers file should make use of the NotImplemented ConnectorError enum variant.
- By doing so, we will throw same error message for all the Connector Implementation 
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2831

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by nain-F49FF806**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Consistent error messages for not implemented payment method.

## Motivation and Context

Resolves  #2861

## How did you test it?
payment connector create
```
{
    "connector_type": "fiz_operations",
    "connector_name": "volt",
    "connector_account_details": {
        "auth_type": "MultiAuthKey",
        "api_key": "{{connector_api_key}}",
        "api_secret": "{{connector_api_secret}}",
        "key1": "{{connector_key1}}",
        "key2": "{{connector_key2}}"
    },
    "test_mode": false,
    "disabled": false,
    "payment_methods_enabled": [
                {
            "payment_method": "bank_redirect",
            "payment_method_types": [
                 {
                    "payment_method_type": "open_banking_uk",
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": false,
                    "installment_payment_enabled": false
                    // "accepted_currencies": {
                    //     "type": "enable_only",
                    //     "list": [
                    //         "EUR"
                    //     ]
                    // },
                    // "accepted_countries": {
                    //     "type": "enable_only",
                    //     "list": [
                    //         "DE"
                    //     ]
                    // }
                }
            ]
        }
    ],
    "metadata": {
        "city": "NY",
        "unit": "245"
    },
    "connector_webhook_details": {
        "merchant_secret": "dce1d767-7c28-429c-b0fe-9e0b9e91f962"
    },
    "business_country": "US",
    "business_label": "food"
}
```

create a payment which is not implemented
Payment create 
```
{
    "amount": 6540,
    "currency": "BRL",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+1",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
     "payment_method": "bank_transfer",
    "payment_method_type": "pix",
    "payment_method_data": {
        "bank_transfer": {
            "pix": {}
        }
    },
    "routing": {
        "type": "single",
        "data": "adyen"
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "BR",
            "first_name": "PiX"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "BR",
            "first_name": "PiX"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}
```

Make any payment for Square for any PM which is not implemented, and see for the error message - it should be payment method not implemented

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by swangi-kumari** [comment]
Hey @nain-F49FF806 , Thanks for PR.
But we'd prefer that contributors comment on an issue before opening a PR for it, so that other contributors are aware that you are working on it. This would also help reduce duplicate effort towards solving the same issue. Please ensure you follow it moving forward.

**Comment by nain-F49FF806** [comment]
@swangi-kumari Noted.

Saw no one seemed assigned, so just went ahead to have a look at the code before making a comment.
I didn't know if I was going to take it, until I had already finished. Sorry. :sweat_smile:  

On a more fortunate note, I only laid eyes on this today, So hopefully not much time wasted. :crossed_fingers:

Regards.

**Comment by swangi-kumari** [comment]
Hey @nain-F49FF806 ,
It's fine if you missed. But can you go to respective issue for which you raised this PR and comment on that issue, so that I can assign it to you.

**Comment by swangi-kumari** [comment]
Hey @nain-F49FF806 ,
Please resolve the conflicts from your branch.

**Comment by nain-F49FF806** [comment]
@swangi-kumari Have rebased onto the most recent `main`. 
Please have a look and let me know if there's anything else needed.

File: crates/router/src/connector/square/transformers.rs [hunk 1]
```diff
@@ -17,19 +17,18 @@ impl TryFrom<(&types::TokenizationRouterData, BankDebitData)> for SquareTokenReq
...
             .into_report(),
 
             BankDebitData::SepaBankDebit { .. }
             | BankDebitData::BecsBankDebit { .. }
-            | BankDebitData::BacsBankDebit { .. } => Err(errors::ConnectorError::NotSupported {
-                message: format!("{:?}", item.request.payment_method_data),
-                connector: "Square",
-            })?,
+            | BankDebitData::BacsBankDebit { .. } => Err(errors::ConnectorError::NotImplemented(
+                utils::get_unimplemented_payment_method_error_message("Square"),
```
Comment (SamraatBansal) [inline crates/router/src/connector/square/transformers.rs:30]: We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

File: crates/router/src/connector/square/transformers.rs [hunk 2]
```diff
@@ -82,40 +81,39 @@ impl TryFrom<(&types::TokenizationRouterData, PayLaterData)> for SquareTokenRequ
...
             | PayLaterData::PayBrightRedirect { .. }
             | PayLaterData::WalleyRedirect { .. }
             | PayLaterData::AlmaRedirect { .. }
-            | PayLaterData::AtomeRedirect { .. } => Err(errors::ConnectorError::NotSupported {
-                message: format!("{:?}", item.request.payment_method_data),
-                connector: "Square",
-            })?,
+            | PayLaterData::AtomeRedirect { .. } => Err(errors::ConnectorError::NotImplemented(
+                utils::get_unimplemented_payment_method_error_message("Square"),
+            ))?,
```
Comment (SamraatBansal) [inline crates/router/src/connector/square/transformers.rs:101]: We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

File: crates/router/src/connector/square/transformers.rs [hunk 3]
```diff
@@ -140,10 +138,9 @@ impl TryFrom<(&types::TokenizationRouterData, WalletData)> for SquareTokenReques
             | WalletData::WeChatPayRedirect(_)
             | WalletData::WeChatPayQr(_)
             | WalletData::CashappQr(_)
-            | WalletData::SwishQr(_) => Err(errors::ConnectorError::NotSupported {
-                message: format!("{:?}", item.request.payment_method_data),
-                connector: "Square",
-            })?,
+            | WalletData::SwishQr(_) => Err(errors::ConnectorError::NotImplemented(
+                utils::get_unimplemented_payment_method_error_message("Square"),
+            ))?,
```
Comment (SamraatBansal) [inline crates/router/src/connector/square/transformers.rs:143]: We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

File: crates/router/src/connector/square/transformers.rs [hunk 4]
```diff
@@ -181,7 +178,7 @@ impl TryFrom<&types::TokenizationRouterData> for SquareTokenRequest {
                 Self::try_from((item, pay_later_data))
             }
             api::PaymentMethodData::GiftCard(_) => Err(errors::ConnectorError::NotImplemented(
-                "Payment Method".to_string(),
+                utils::get_unimplemented_payment_method_error_message("Square"),
```
Comment (SamraatBansal) [inline crates/router/src/connector/square/transformers.rs:181]: We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

File: crates/router/src/connector/square/transformers.rs [hunk 5]
```diff
@@ -298,7 +294,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for SquarePaymentsRequest {
             | api::PaymentMethodData::GiftCard(_)
             | api::PaymentMethodData::PayLater(_)
             | api::PaymentMethodData::Wallet(_) => Err(errors::ConnectorError::NotImplemented(
-                "Payment Method".to_string(),
+                utils::get_unimplemented_payment_method_error_message("Square"),
```
Comment (SamraatBansal) [inline crates/router/src/connector/square/transformers.rs:297]: We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/square/transformers.rs (hunk 1 above)
Comment (nain-F49FF806) [reply crates/router/src/connector/square/transformers.rs:30]: I had assumed the two blocks were kept separate on purpose to have the `AchBank` arm generate a report, and the others not. 

On second look, it seems the conversion happens automatically for the others anyway, to keep the return type identical. 

I shall club them all and also add the `.into_report` at the end. Let me know if that is not as intended.

File: crates/router/src/connector/square/transformers.rs (hunk 5 above)
Comment (nain-F49FF806) [reply crates/router/src/connector/square/transformers.rs]: Done

**Review by SamraatBansal — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Pull request merged**